articles/ajax-request-in-django-using-axios/index.md

```diff
@@ -0,0 +1,246 @@
+
+```
+
+If you get an error message command, run the command below to install `virtualenv` on your computer.
+
+```bash
+
+ pip install virtualenv
+
+```
+
+Change to any directory of your choice on your system/laptop or you can as well make a new directory.  Create a virtual environment for our project using the `virtualenv`.
+
+ Run the command below to create and activate the virtual environment.
+
+```bash
+
+$ virtualenv venv
+
+$ source venv/bin/activate
+
+```
+
+ 
+
+I have created a starter template for you to follow along in this tutorial, clone the starter template in your current directory using the command below.
+
+```bash 
+
+ git clone https://github.com/atoyegbe/note-app.git
+
+ ```
+
+Run the command below to install the dependency needed for this project to work.
+
+```bash 
+
+pip install -r requirements.txt
+
+```
+
+Let’s make sure nothing is broken in our application, change into the project directory and run the command below.
+
+```bash 
+python manage.py runserver
+```
+After running the above command you will see the below image in your browser.
+![form page](/engineering-education/ajax-request-in-django-using-axios/form_page.png)
+
+Let’s go into the **templates/home.html** and we will add to our project the **Axios** library by adding the CDN script tag below outside the div tag.
+
+```html
+
+{% include ‘base.html’ %}
+
+{% block content %}
+
+<div class="container">
+
+     <h1> Notes </h1>
+
+ .....
+
+ ....
+
+</div>
+
+<!-- Adding axios script tag here-->
+
+<script src="https://cdnjs.cloudflare.com/ajax/libs/axios/0.21.1/axios.min.js"></script>
+
+{% endblock %}
+
+```
+
+### Using Axios to submit a form.
+
+Adding event listener to the form and sending form data to Django serverside. Add these code snippets below the Axios script tag.
+
+```javascript
+<script>
+let form = document.getElementById(‘form’); #selecting the form
+
+form.addEventListener(‘submit’, function(event) { #a
+    event.preventDefault()
+    
+    let data = new FormData(); #b
+    
+    data.append(“title”, document.getElementById(‘title’).value)  
+    data.append(“note”, document.getElementById(‘note’).value)
+    data.append(“csrfmiddlewaretoken”, ‘{{csrf_token}}’) #c
+    
+    axios.post(‘create_note/’, data) #d
+     .then(res => alert(“Form Submitted”)) #e
+     .catch(errors => console.log(errors)) #f
+
+})
+
+</script>
+
+```
+The above code 
+   - #a - Adding a submit event listener to the form.
+   - #b - Creating a new Form in JavaScript.
+   - #c - adding a CRSF token, if we do not include this we get a 403 forbidden response we won’t be able to submit the form data.
+   - #d - Here we use **axios.post** method to submit form data.
+   - #e - A alert message **Form Submitted** pop on your windows if the form submitted successfully.
+   - #f - This catches and displays the error in the console if an error occurs when submitting the form.
+
+Your **home.html** file should exactly like this.
+
```

Comment:
There are some missing words. Edit removing 'exactly' to look like this: 'Your `home.html` should look like this.'